app/src/main/java/com/example/danil/androidcourses/implicitintent/ImplicitActivity.java

```diff
@@ -0,0 +1,46 @@
+package com.example.danil.androidcourses.implicitintent;
+
+import android.content.Intent;
+import android.net.Uri;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+
+import com.example.danil.androidcourses.R;
+
+public class ImplicitActivity extends AppCompatActivity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_implicit_intent);
+
+        Button btnWeb = (Button)findViewById(R.id.btn_web);
+        btnWeb.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse("http://www.google.com"));
+                startActivity(intent);
+            }
+        });
+
+        Button btnCall = (Button)findViewById(R.id.btn_call);
+        btnCall.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse("tel:0958050679"));
+                startActivity(intent);
+            }
+        });
+
+        Button btnMap = (Button)findViewById(R.id.btn_map);
+        btnMap.setOnClickListener(new View.OnClickListener() {
```

Comment:
Lets use Java 8 lambdas for setOnClickListener. Use that links, which I have sent you.